v2/tracing.go

```diff
@@ -3,42 +3,76 @@ package shuttle
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
-	"github.com/Azure/go-shuttle/v2/otel"
+	shuttleotel "github.com/Azure/go-shuttle/v2/otel"
 	"go.opentelemetry.io/otel/trace"
 )
 
+const (
+	serviceTracerName             = "go-shuttle"
+	defaultReceiverHandleSpanName = "receiver.Handle"
+)
+
 type TracingHandlerOpts struct {
-	enrichSpan func(context.Context, trace.Span)
+	spanStartOptions []trace.SpanStartOption
+	traceProvider    trace.TracerProvider
+	spanNameFormat   func(defaultSpanName string, message *azservicebus.ReceivedMessage) string
+}
+
+func (t *TracingHandlerOpts) tracer() trace.Tracer {
+	if t.traceProvider == nil {
+		return otel.Tracer(serviceTracerName)
+	}
+	return t.traceProvider.Tracer(serviceTracerName)
 }
 
 // NewTracingHandler is a shuttle middleware that extracts the context from the message Application property if available,
 // or from the existing context if not, and starts a span.
 func NewTracingHandler(next Handler, options ...func(t *TracingHandlerOpts)) HandlerFunc {
-	t := &TracingHandlerOpts{}
+	t := &TracingHandlerOpts{
+		spanNameFormat: func(defaultSpanName string, message *azservicebus.ReceivedMessage) string {
+			return defaultSpanName
+		},
+	}
 	for _, opt := range options {
 		opt(t)
 	}
 	return func(ctx context.Context, settler MessageSettler, message *azservicebus.ReceivedMessage) {
-		ctx, span := otel.Extract(ctx, message)
+		defaultStartOptions := []trace.SpanStartOption{trace.WithAttributes(shuttleotel.MessageAttributes(message)...)}
```

Comment:
shuttleotel.MessageAttributes is what extracts message attributes and puts it on the span.